airflow/ui/src/pages/Run/Header.tsx

```diff
@@ -36,44 +34,39 @@ export const Header = ({
   readonly dagRun: DAGRunResponse;
   readonly isRefreshing?: boolean;
 }) => (
-  <Box borderColor="border" borderRadius={8} borderWidth={1} p={2}>
-    <Flex alignItems="center" justifyContent="space-between" mb={2}>
-      <HStack alignItems="center" gap={2}>
-        <FiBarChart size="1.75rem" />
-        <Heading size="lg">
-          <strong>Run: </strong>
-          {dagRun.dag_run_id}
-        </Heading>
-        <StateBadge state={dagRun.state}>{dagRun.state}</StateBadge>
-        {isRefreshing ? <Spinner /> : <div />}
-      </HStack>
-      <HStack>
-        {dagRun.note === null || dagRun.note.length === 0 ? undefined : (
-          <DisplayMarkdownButton
-            header="Dag Run Note"
-            icon={<FiMessageSquare color="black" />}
-            mdContent={dagRun.note}
-            text="Note"
-          />
-        )}
+  <HeaderCard
+    actions={
+      <>
         <ClearRunButton dagRun={dagRun} />
         <MarkRunAsButton dagRun={dagRun} />
-      </HStack>
-    </Flex>
-    <SimpleGrid columns={4} gap={4}>
-      <Stat label="Run Type">
-        <HStack>
-          <RunTypeIcon runType={dagRun.run_type} />
-          <Text>{dagRun.run_type}</Text>
-        </HStack>
-      </Stat>
-      <Stat label="Start">
-        <Time datetime={dagRun.start_date} />
-      </Stat>
-      <Stat label="End">
-        <Time datetime={dagRun.end_date} />
-      </Stat>
-      <Stat label="Duration">{getDuration(dagRun.start_date, dagRun.end_date)}s</Stat>
-    </SimpleGrid>
-  </Box>
+      </>
+    }
+    icon={<FiBarChart />}
+    isRefreshing={isRefreshing}
+    note={dagRun.note}
+    state={dagRun.state}
+    stats={[
+      ...(dagRun.logical_date === null
+        ? []
+        : [
+            {
+              label: "Logical Date",
+              value: <Time datetime={dagRun.logical_date} />,
+            },
+          ]),
+      {
+        label: "Run Type",
+        value: (
+          <HStack>
+            <RunTypeIcon runType={dagRun.run_type} />
+            <Text>{dagRun.run_type}</Text>
+          </HStack>
+        ),
+      },
+      { label: "Start", value: <Time datetime={dagRun.start_date} /> },
+      { label: "End", value: <Time datetime={dagRun.end_date} /> },
+      { label: "Duration", value: `${getDuration(dagRun.start_date, dagRun.end_date)}s` },
```

Comment:
I don't think date should wrap like this when we have space to inline them.
![Screenshot 2025-02-24 at 18 06 14](https://github.com/user-attachments/assets/633532d9-15e7-42b4-ab56-77cc691c2f3e)

Reply:
Will update in a later pr